bfvmm/src/vmcs/test/test_vmcs_intel_x64.cpp

```diff
@@ -193,22 +448,38 @@ setup_vmcs_x64_state_intrinsics(MockRepository &mocks, vmcs_intel_x64_state *sta
 }
 
 static void
+setup_vmcs_launch_failure(MockRepository &mocks, intrinsics_intel_x64 *in)
+{
+    setup_msrs();
+    setup_vmcs_fields();
+
+    Call &vmlaunch = mocks.OnCall(in, intrinsics_intel_x64::vmlaunch).Return(false);
+    mocks.OnCall(in, intrinsics_intel_x64::read_msr).After(vmlaunch).Do(read_msr);
```

Comment:
since your calling read_msr, which was previously setup, what exactly is the reason for this code? I would prefer to see a flag in the read_msr code that you set saying something like `return_crap=true`